plugin.video.nfl.gamepass/default.py

```diff
@@ -0,0 +1,635 @@
+# -*- coding: utf-8 -*-
+"""
+A Kodi addon/skin for NFL Game Pass
+"""
+
+import sys
+import json
+import urllib.request, urllib.parse, urllib.error
+from traceback import format_exc
+from datetime import timedelta
+import logging
+import time
+
+import xbmc
+import xbmcaddon
+import xbmcgui
+import xbmcvfs
+
+from resources.lib.pigskin.pigskin import pigskin
+from resources.lib import kodilogging
+
+dialog = xbmcgui.Dialog()
+
+# Addon Init
+addon = xbmcaddon.Addon()
+language = addon.getLocalizedString
+ADDON_PATH = xbmc.translatePath(addon.getAddonInfo('path'))
+ADDON_PROFILE = xbmc.translatePath(addon.getAddonInfo('profile'))
+
+# Default Window IDs
+WINDOW_FULLSCREEN_VIDEO = 12005
+
+# Logging Init
+logger = logging.getLogger(addon.getAddonInfo('id'))
+kodilogging.config()
+
+# Create addon_data directory
+if not xbmcvfs.exists(ADDON_PROFILE):
+    xbmcvfs.mkdir(ADDON_PROFILE)
+
+
+def get_credentials():
+    username = addon.getSetting('email')
+    password = addon.getSetting('password')
+
+    if not username or not password:
+        answer = dialog.ok(language(30021), language(30050))
+
+        if answer:
+            addon.openSettings()
+            username = addon.getSetting('email')
+            password = addon.getSetting('password')
+        else:
+            sys.exit(0)
+
+    return username, password
+
+
+def build_proxy_url():
+    try:
+        protocol = addon.getSetting('proxy_scheme') + '://'
+
+        auth = ''
+        if addon.getSetting('proxy_auth') == 'true':
+            username = addon.getSetting('proxy_username').strip()
+            password = addon.getSetting('proxy_password')
+
+            if not username or not password:
+                return ''
+
+            auth = '%s:%s@' % (urllib.parse.quote(username), urllib.parse.quote(password))
+
+        host = addon.getSetting('proxy_host').strip()
+        if not host:
+            return ''
+
+        port = addon.getSetting('proxy_port').strip()
+        if port:
+            host = host + ':' + port
+
+        proxy_url = protocol + auth + host
+
+        return proxy_url
+    except Exception as e:
+        logger.debug('Failed building proxy url, using none')
+        logger.exception(e)
+        return None
+
+
+def has_inputstream_adaptive():
+    """Checks if InputStream Adaptive is installed and enabled."""
+    payload = {
+        'jsonrpc': '2.0',
+        'id': 1,
+        'method': 'Addons.GetAddonDetails',
+        'params': {
+            'addonid': 'inputstream.adaptive',
+            'properties': ['enabled']
+        }
+    }
+    response = xbmc.executeJSONRPC(json.dumps(payload))
```

Comment:
Why not use `xbmc.getCondVisibility('System.HasAddon("inputstream.adaptive")')`. It will return `false` if not enabled.